### PR TITLE
Close #1 Create Merchants record endpoints

### LINKED ISSUE #1: Create Merchants record endpoints
Merchants record endpoints

### PR DESCRIPTION
**Pull request by GSmes**
@concach
- Add namespaced routes to routes.rb
- Create Merchant model, migration, and controllers

### CODE REVIEW

File: app/controllers/api/v1/merchants/random_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Api::V1::Merchants::RandomController < ApiBaseController
+  respond_to :json
+
+  def show
+    respond_with Merchant.order("RANDOM()").first
```
Comment (cconcannon) [inline app/controllers/api/v1/merchants/random_controller.rb:5]: do we still need to assign the instance variable @merchant? I know it won't be used for any view, but I'm writing my controller tests to take advantage of the assigning of instance variables... we'll see when we go over testing tomorrow

File: app/controllers/api/v1/merchants/search_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class Api::V1::Merchants::SearchController < ApiBaseController
+  respond_to :json
+
+  def index
+    respond_with Merchant.where(merchant_params)
```
Comment (cconcannon) [inline app/controllers/api/v1/merchants/search_controller.rb:5]: Again, might need a @merchants = Merchant.all ... we'll see further down the road

**Comment by cconcannon** [comment]
good work!

**Comment by GSmes** [comment]
@concach Yeah, I wasn't entirely sure about that. For now I'm not using instance variables, but it's an easy change if necessary.

**Pull request merged**